### PR TITLE
[sdk] update to GSDK 4.4.0

### PR DESCRIPTION
**Pull request by lmnotran**
This updates the repo to work with the [Gecko SDK v4.4.0](https://github.com/SiliconLabs/gecko_sdk/releases/tag/v4.4.0) release

---
# Other changes

- Update scripts to install ARM GNU Toolchain 12
  - `arm-none-eabi-gcc 12` is a requirement of GSDK v4.4.0
  - Removed older toolchains from `Build` workflow
- Add ot_stack_features_config component to ot-efr32 extension
  - This adds a new component that generates a config file for `SL_OPENTHREAD_STACK_FEATURES_CONFIG_FILE`
  - This is essentially a copy of the ot_stack_ftd/mtd/rcp components except it doesn't define a `OPENTHREAD_<DEVICE_TYPE>` macro
https://github.com/SiliconLabs/gecko_sdk/blob/124fa19de8c8b3961d21c20857f7df32239786da/protocol/openthread/component/ot_stack_ftd.slcc#L21-L26
  - The component also uses the [validation script](https://github.com/SiliconLabs/gecko_sdk/blob/gsdk_4.4/protocol/openthread/component/script/ot_log_validation.lua) to ensure [`ot_rtt_log`](https://github.com/SiliconLabs/gecko_sdk/blob/gsdk_4.4/protocol/openthread/component/ot_rtt_log.slcc) is in a project if the `OPENTHREAD_CONFIG_LOG_OUTPUT` is set to `OPENTHREAD_CONFIG_LOG_PLATFORM_DEFINED`
- Update the `openthread` submodule to the head of `main`

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -50,18 +50,12 @@ jobs:
...
-          - gcc_ver: 9
-            gcc_download_url: https://developer.arm.com/-/media/Files/downloads/gnu-rm/9-2019q4/RC2.1/gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2
-            gcc_extract_dir: gcc-arm-none-eabi-9-2019-q4-major
-          - gcc_ver: 10.3
-            gcc_download_url: https://developer.arm.com/-/media/Files/downloads/gnu-rm/10.3-2021.10/gcc-arm-none-eabi-10.3-2021.10-x86_64-linux.tar.bz2
-            gcc_extract_dir: gcc-arm-none-eabi-10.3-2021.10
+          - gcc_ver: 12.2.Rel1
+            gcc_download_url: https://developer.arm.com/-/media/Files/downloads/gnu/12.2.rel1/binrel/arm-gnu-toolchain-12.2.rel1-x86_64-arm-none-eabi.tar.xz
+            gcc_extract_dir: arm-gnu-toolchain-12.2.rel1-x86_64-arm-none-eabi
+          - gcc_ver: 12.3.Rel1
```
Comment (romacdon) [inline .github/workflows/build.yml:56]: Why do we include 12.3?

File: examples/example_vendor_slc_extension/projects/example-vendor-soc-with-buttons.slcp [hunk 2]
```diff
@@ -36,7 +36,7 @@ quality: production
 component:
   - id: ot_platform_abstraction_core
   - id: ot_psa_crypto
-  - id: ot_thirdparty
+  - id: ot_mbedtls
   - id: uartdrv_usart
```
Comment (romacdon) [inline examples/example_vendor_slc_extension/projects/example-vendor-soc-with-buttons.slcp:40]: Why doesn't this slcp include the following which was added to the other slcps.

 - id: ot_stack_features_config
    from: ot-efr32

File: script/bootstrap [hunk 3]
```diff
@@ -118,6 +119,38 @@ do_bootstrap_silabs()
     echo "Bootstrapping silabs"
     "${repo_dir}"/script/bootstrap_silabs
 }
+
+install_arm_toolchain()
+{
+    local url=${1-"https://developer.arm.com/-/media/Files/downloads/gnu/12.3.rel1/binrel/arm-gnu-toolchain-12.3.rel1-x86_64-arm-none-eabi.tar.xz"}
```
Comment (romacdon) [inline script/bootstrap:125]: Should this be 12.2?

File: src/src/radio.c [hunk 4]
```diff
@@ -1643,11 +1713,8 @@ otError otPlatRadioTransmit(otInstance *aInstance, otRadioFrame *aFrame)
                             + SHR_DURATION_US);
         // Note - we need to call this outside of txCurrentPacket as for Series 2,
         // this results in calling the SE interface from a critical section which is not permitted.
-
 #if OPENTHREAD_RADIO && OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE == 1
         radioProcessTransmitSecurity(sTxFrame, sTxFrame->mIid);
-#else
-        radioProcessTransmitSecurity(sTxFrame, 0);
```
Comment (romacdon) [inline src/src/radio.c:1650]: The GSDK radio.c file does not have the #if check around the call to radioProcessTransmitSecurity.  Here's the snippet of text from the GSDK version.

```
        updateIeInfoTxFrame(sTxFrame->mInfo.mTxInfo.mTxDelayBaseTime + sTxFrame->mInfo.mTxInfo.mTxDelay + SHR_DURATION_US);
        // Note - we need to call this outside of txCurrentPacket as for Series 2,
        // this results in calling the SE interface from a critical section which is not permitted.
        radioProcessTransmitSecurity(sTxFrame, sTxFrame->mIid);
#endif // OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2

```

File: src/src/radio.c [hunk 5]
```diff
@@ -3053,58 +3149,66 @@ static bool validatePacketTimestamp(RAIL_RxPacketDetails_t *pPacketDetails, uint
...
     {
-        sReceiveAckFrame.mInfo.mRxInfo.mRssi = pPacketDetails->rssi;
-        sReceiveAckFrame.mInfo.mRxInfo.mLqi  = pPacketDetails->lqi;
-        sReceiveAckFrame.mInfo.mRxInfo.mTimestamp =
-            pPacketDetails->timeReceived.packetTime + ((uint64_t)railUsTimerWraps << 32);
-// TODO: Maybe remove #if ?
+        sReceiveAckFrame.mInfo.mRxInfo.mRssi      = pPacketDetails->rssi;
+        sReceiveAckFrame.mInfo.mRxInfo.mLqi       = pPacketDetails->lqi;
+        sReceiveAckFrame.mInfo.mRxInfo.mTimestamp = timestamp + ((uint64_t)railUsTimerWraps << 32);
 #if OPENTHREAD_RADIO && OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE == 1
```
Comment (romacdon) [inline src/src/radio.c:3176]: GSDK version does not have #if wrapper around   `sReceiveAckFrame.mIid = iid; `

File: src/src/radio.c [hunk 6]
```diff
@@ -3053,58 +3149,66 @@ static bool validatePacketTimestamp(RAIL_RxPacketDetails_t *pPacketDetails, uint
...
+        }
+
+        sReceiveFrame.mInfo.mRxInfo.mTimestamp = timestamp + ((uint64_t)railUsTimerWraps << 32);
         // Set this flag only when the packet is really acknowledged with a secured enhanced ACK.
-        sReceiveFrame[sReceiveBufferInUse].mInfo.mRxInfo.mAckedWithSecEnhAck = securedOutgoingEnhancedAck;
+        sReceiveFrame.mInfo.mRxInfo.mAckedWithSecEnhAck = securedOutgoingEnhancedAck;
         // Set this flag only when the packet is really acknowledged with frame pending set.
-        sReceiveFrame[sReceiveBufferInUse].mInfo.mRxInfo.mAckedWithFramePending = framePendingSetInOutgoingAck;
+        sReceiveFrame.mInfo.mRxInfo.mAckedWithFramePending = framePendingSetInOutgoingAck;
 #if OPENTHREAD_RADIO && OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE == 1
```
Comment (romacdon) [inline src/src/radio.c:3204]: GSDK version does not have #if wrapper around   `sReceiveFrame.mIid = iid; `

**Review by romacdon — CHANGES_REQUESTED** [review]
A few changes are needed.

File: src/src/radio.c (hunk 4 above)
Comment (romacdon) [reply src/src/radio.c:1650]: I see. Thanks.

File: src/src/radio.c (hunk 5 above)
Comment (romacdon) [reply src/src/radio.c:3176]: You can ignore this comment now.  I understand why there is a required difference.

File: examples/example_vendor_slc_extension/projects/example-vendor-soc-with-buttons.slcp (hunk 2 above)
Comment (lmnotran) [reply examples/example_vendor_slc_extension/projects/example-vendor-soc-with-buttons.slcp:40]: Good catch!

File: .github/workflows/build.yml (hunk 1 above)
Comment (lmnotran) [reply .github/workflows/build.yml:56]: It's a newer minor version and I imagine people might want to use it. I don't think we should remove it

File: src/src/radio.c (hunk 6 above)
Comment (romacdon) [reply src/src/radio.c:3204]: You can ignore this comment now. I understand why there is a required difference.

**Review by romacdon — APPROVED** [review]
LGTM

**Comment by lmnotran** [comment]
Hi @jwhui !

Can you remove the requirement for the `arm-gcc-6/7/8/9/10.3` checks and replace them with the new `arm-gcc-12.2.Rel1/12.3.Rel1` checks?

Also after this PR is merged, we should be safe to continue approving the dependabot PRs.

Thanks!

**Review by hemanth-silabs — APPROVED** [review]
LGTM 👍

**Review by bertoldi-silabs — APPROVED** [review]

**Review by parag-silabs — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**